From 47934f00dfe415af7f83bd781ce3d920eaaaf248 Mon Sep 17 00:00:00 2001
From: Joe Zhou <shjzhou@marvell.com>
Date: Thu, 20 Feb 2014 07:08:09 +0800
Subject: [PATCH 1409/1825] alp: support all 7 ALP A0 flavors

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit f483f3c448b4f6e3f2db4e43263985f13de96481

	new flavors are added in function mv_sys_map

Signed-off-by: Joe Zhou <shjzhou@marvell.com>

Change-Id: Ie3aaae0f43f109afe86bf852345f3511d8177b25
Reviewed-on: http://vgitil04.il.marvell.com:8080/5831
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Yehuda Yitschak <yehuday@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 .../avanta_lp_family/ctrlEnv/sys/mvAhbToMbus.c     |   10 ++++++++++
 arch/arm/mach-avantalp/sysmap.c                    |    4 ++++
 2 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/sys/mvAhbToMbus.c b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/sys/mvAhbToMbus.c
index ccddfcf..883d4f5 100644
--- a/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/sys/mvAhbToMbus.c
+++ b/arch/arm/mach-avantalp/avanta_lp_family/ctrlEnv/sys/mvAhbToMbus.c
@@ -691,7 +691,12 @@ static MV_STATUS ahbToMbusRemapRegOffsGet(MV_U32 winNum, AHB_TO_MBUS_REMAP_REG_O
 			(ctrlMode == MV_78260_DEV_ID) ||
 			(ctrlMode == MV_78460_DEV_ID) ||
 			(ctrlMode == MV_6660_DEV_ID) ||
+			(ctrlMode == MV_6665_DEV_ID) ||
 			(ctrlMode == MV_6650_DEV_ID) ||
+			(ctrlMode == MV_6650F_DEV_ID) ||
+			(ctrlMode == MV_6658_DEV_ID) ||
+			(ctrlMode == MV_6610_DEV_ID) ||
+			(ctrlMode == MV_6610F_DEV_ID) ||
 			(ctrlMode == MV_78000_DEV_ID)) {
 			pRemapRegs->lowRegOffs = AHB_TO_MBUS_WIN_REMAP_LOW_REG(winNum);
 			pRemapRegs->highRegOffs = AHB_TO_MBUS_WIN_REMAP_HIGH_REG(winNum);
@@ -719,7 +724,12 @@ static MV_STATUS ahbToMbusRemapRegOffsGet(MV_U32 winNum, AHB_TO_MBUS_REMAP_REG_O
 			(ctrlMode == MV_78260_DEV_ID) ||
 			(ctrlMode == MV_78460_DEV_ID) ||
 			(ctrlMode == MV_6660_DEV_ID) ||
+			(ctrlMode == MV_6665_DEV_ID) ||
 			(ctrlMode == MV_6650_DEV_ID) ||
+			(ctrlMode == MV_6650F_DEV_ID) ||
+			(ctrlMode == MV_6658_DEV_ID) ||
+			(ctrlMode == MV_6610_DEV_ID) ||
+			(ctrlMode == MV_6610F_DEV_ID) ||
 			(ctrlMode == MV_78000_DEV_ID)) {
 			pRemapRegs->lowRegOffs = AHB_TO_MBUS_WIN_REMAP_LOW_REG(winNum);
 			pRemapRegs->highRegOffs = AHB_TO_MBUS_WIN_REMAP_HIGH_REG(winNum);
diff --git a/arch/arm/mach-avantalp/sysmap.c b/arch/arm/mach-avantalp/sysmap.c
index 043dd02..6d9cf1a 100644
--- a/arch/arm/mach-avantalp/sysmap.c
+++ b/arch/arm/mach-avantalp/sysmap.c
@@ -156,12 +156,16 @@ MV_CPU_DEC_WIN *mv_sys_map(void)
 		map = SYSMAP_ALP_6660;
 		break;
 	case MV_6660_DEV_ID:
+	case MV_6665_DEV_ID:
 		map = SYSMAP_ALP_6660;
 		break;
 	case MV_6650_DEV_ID:
+	case MV_6650F_DEV_ID:
+	case MV_6658_DEV_ID:
 		map = SYSMAP_ALP_6650;
 		break;
 	case MV_6610_DEV_ID:
+	case MV_6610F_DEV_ID:
 		map = SYSMAP_ALP_6610;
 		break;
 	default:
-- 
1.7.5.4

